src/mlpack/methods/ann/ffn.hpp

```diff
@@ -382,16 +386,26 @@ class FFN
                   const TargetsType& targets,
                   GradientsType& gradients);
 
- private:
-  // Helper functions.
   /**
-   * The Forward algorithm (part of the Forward-Backward algorithm).  Computes
-   * forward probabilities for each module.
+   * Set the logical dimensions of the input.
    *
-   * @param input Data sequence to compute probabilities for.
+   * TODO: better comment.  You would call this when you want to, e.g., pass an
+   * n-dimensional tensor, so that you can specify each of those n dimensions.
    */
-  template<typename InputType>
-  void Forward(const InputType& input);
+  // Note: we don't need to invalidate any caches, because any forward pass will
+  // already check if the input dimensions have changed.
+  std::vector<size_t>& InputDimensions() { return inputDimensions; }
+  //! Get the logical dimensions of the input.
+  const std::vector<size_t>& InputDimensions() const
+  {
+    // The user may change the input dimensions, so we will have to propagate
+    // these changes to the network.
+    inputDimensionsAreSet = false;
+    return inputDimensions;
+  }
+
```

Comment:
I think there is an issue with this
I think you should interchange when we have to set inputDimensionsAreSet = false

Reply:
Yes, the non-`const` version should be the one where we set `inputDimensionsAreSet` to `false`.  Nice catch!  Thank you.  Fixed in 9a71a61aa.